spec/async/http/cache/general_spec.rb

```diff
@@ -70,43 +70,112 @@
 				else
 					Protocol::HTTP::Response[200, [['cache-control', 'max-age=1, public'], ['vary', 'user-agent']], ['Hello', ' ', 'World']]
 				end
-				
+
 				if request.head?
 					response.body = Protocol::HTTP::Body::Head.for(response.body)
 				end
-				
+
 				response
 			end
 		end
-		
+
 		let(:user_agents) {[
 			'test-a',
 			'test-b',
 		]}
-		
+
 		it "should cache GET requests" do
 			2.times do
 				user_agents.each do |user_agent|
 					response = subject.get("/", {'user-agent' => user_agent})
 					expect(response.headers['vary']).to include('user-agent')
-					expect(response.read).to be == user_agent
+					expect(response.read).to eq(user_agent)
 				end
 			end
-			
-			expect(store.index.size).to be 2
+
+			expect(store.index.size).to eq(2)
 		end
-	end
-	
+  end
+
+  context 'cache writes' do
+    context 'by response code' do
+      let(:app) do
+        Protocol::HTTP::Middleware.for do |_request|
+          Protocol::HTTP::Response[response_code, [], ['body']]
```

Comment:
@ioquatix This is the test where you suggested I rebase and try again. I have now, but I'm still getting a timeout here when `body` isn't empty:
```
    cache writes
      by response code
        when cacheable: 200
 0.19s     warn: Async::HTTP::Client [oid=0x1bbc] [ec=0x1bd0] [pid=14740] [2022-08-22 19:32:36 -0700]
               | Waiting for Async::HTTP::Protocol::HTTP1 pool to drain: #<Async::Pool::Controller(2/∞) 1/1/1;1/1/1>
          is cached (FAILED - 1)
        when cacheable: 203
10.18s     warn: Async::HTTP::Client [oid=0x1d74] [ec=0x1d88] [pid=14740] [2022-08-22 19:32:46 -0700]
               | Waiting for Async::HTTP::Protocol::HTTP1 pool to drain: #<Async::Pool::Controller(2/∞) 1/1/1;1/1/1>
          is cached (FAILED - 2)
...
```

Reply:
I'll check what's going on.